### DISCUSSION TITLE
What is a pl way to reload dataloader?

### DISCUSSION THREAD

**Discussion by Vozf** [Lightning-AI/pytorch-lightning#7012] [General]
I would like to change dataset parameters(e.g. other augmentations). What is the pl way to reinit dataset and dataloader with new parameters?

**Comment by ananthsub** [comment]
At what points during training do you need to change the dataset parameters? if it's at the start of every epoch, this flag should work for you: https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html#reload-dataloaders-every-epoch

If you're using a lightning datamodule, you can re-initialize the dataset and then create a new dataloader across training/validation/test

**Reply by isabellahuang** [reply]
How exactly do we re-initialize the dataset every epoch, e.g. to load new unseen data? I don't see a way to trigger `prepare_data` at the start of every epoch.

**Reply by carmocca** [reply]
You can use `Trainer(reload_dataloaders_every_epoch)` to recreate them.

But `prepare_data` is meant to download/preprocess the data only once before training, it's not meant to be called every epoch
